libcurl_vendor/CMakeLists.txt

```diff
@@ -14,57 +14,40 @@ if (POLICY CMP0135)
 endif()
 
 macro(build_libcurl)
+
   set(extra_cxx_flags)
   set(extra_c_flags)
   if(NOT WIN32)
-    list(APPEND extra_cxx_flags "-std=c++14 -w")
+    list(APPEND extra_cxx_flags "-w")
     list(APPEND extra_c_flags "-w")
   endif()
 
-  set(PKGCONFIG_FOR_OPENSSL)
-  if(APPLE)
-    set(PKGCONFIG_FOR_OPENSSL "/usr/local/opt/openssl/lib/pkgconfig")
-  endif()
-
   if(WIN32 AND NOT ${CMAKE_VERBOSE_MAKEFILE})
     set(should_log ON)  # prevent warnings in Windows CI
   else()
     set(should_log OFF)
   endif()
 
   include(ExternalProject)
-  if(WIN32)
-    ExternalProject_Add(curl-7.81.0
-      URL https://github.com/curl/curl/releases/download/curl-7_81_0/curl-7.81.0.tar.gz
-      URL_MD5 9e5e81fc7657eea8dc66672768082c46
-      LOG_CONFIGURE ${should_log}
-      LOG_BUILD ${should_log}
-      CMAKE_ARGS
-        -DENABLE_MANUAL=OFF
-        -DBUILD_TESTING=OFF
-        -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/libcurl_install
-        -Wno-dev
-      TIMEOUT 600
-    )
-  else()
-    ExternalProject_Add(curl-7.81.0
-      URL https://github.com/curl/curl/releases/download/curl-7_81_0/curl-7.81.0.tar.gz
-      URL_MD5 9e5e81fc7657eea8dc66672768082c46
-      CONFIGURE_COMMAND
-        <SOURCE_DIR>/configure
-        CFLAGS=${extra_c_flags}
-        CXXFLAGS=${extra_cxx_flags}
-        PKG_CONFIG_PATH=$PKG_CONFIG_PATH:${PKGCONFIG_FOR_OPENSSL}
-        --prefix=${CMAKE_CURRENT_BINARY_DIR}/libcurl_install
-        --with-ssl
-        --without-libgsasl
-        --without-libpsl
-        --without-libidn2
-      BUILD_COMMAND $(MAKE)
-      INSTALL_COMMAND $(MAKE) install
-      TIMEOUT 600
-    )
-  endif()
+
+  ExternalProject_Add(curl-7.81.0
+    URL https://github.com/curl/curl/releases/download/curl-7_81_0/curl-7.81.0.tar.gz
+    URL_MD5 9e5e81fc7657eea8dc66672768082c46
+    LOG_CONFIGURE ${should_log}
+    LOG_BUILD ${should_log}
+    CMAKE_ARGS
+      -DCMAKE_CXX_STANDARD=14
+      -DCMAKE_CXX_FLAGS=${extra_cxx_flags}
+      -DCMAKE_C_FLAGS=${extra_c_flags}
+      -DENABLE_MANUAL=OFF
+      -DBUILD_TESTING=OFF
+      -DCURL_ENABLE_SSL=ON #default, but just to make clear
+      -DCURL_USE_LIBPSL=OFF
+      -DUSE_LIBIDN2=OFF
+      -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/libcurl_install
+      -Wno-dev
```

Comment:
It is probably worthwhile to drop this and see what happens.  It shouldn't be necessary in most cases.

Reply:
I think if we remove line 47 we'll need to update the install statement that comes right after the `ExternalProject_Add`. Could we just get rid of that whole statement and set CMAKE_INSTALL_PREFIX to something like `${CMAKE_INSTALL_PREFIX}/opt/libcurl_vendor`?

Reply:
Oh, I meant the `-Wno-dev` here; I don't think we need that.

Reply:
I just tried omitting the `CMAKE_INSTALL_PREFIX` altogether and I get this when trying to build:
```
[1/2] Performing install step for 'curl-7.81.0'
FAILED: curl-7.81.0-prefix/src/curl-7.81.0-stamp/curl-7.81.0-install /workspaces/raptor/playground/ros2ws/build/libcurl_vendor/curl-7.81.0-prefix/src/curl-7.81.0-stamp/curl-7.81.0-install 
cd /workspaces/raptor/playground/ros2ws/build/libcurl_vendor/curl-7.81.0-prefix/src/curl-7.81.0-build && /usr/local/bin/cmake --build . --target install && /usr/local/bin/cmake -E touch /workspaces/raptor/playground/ros2ws/build/libcurl_vendor/curl-7.81.0-prefix/src/curl-7.81.0-stamp/curl-7.81.0-install
[0/1] Install the project...
-- Install configuration: ""
-- Installing: /usr/local/lib/libcurl.so
CMake Error at lib/cmake_install.cmake:52 (file):
  file INSTALL cannot copy file
  "/workspaces/raptor/playground/ros2ws/build/libcurl_vendor/curl-7.81.0-prefix/src/curl-7.81.0-build/lib/libcurl.so"
  to "/usr/local/lib/libcurl.so": Success.
Call Stack (most recent call first):
  cmake_install.cmake:47 (include)
```

naively it looks like I just need to forward the `CMAKE_INSTALL_PREFIX` to the `ExternalProject_Add` call.

Reply:
> Oh, I meant the `-Wno-dev` here; I don't think we need that.

Ah...sorry about that.  I've added the option for turning libidn2 off.  I can also revert my chance on the CMAKE_INSTALL_PREFIX as well if it's needed.  It seems a little odd to me to install it in your build directory just to copy it to it's final destination....so I think how it is now is a little cleaner, but there could be reasons I'm not aware of for it being as it was.